src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/ManifestReader.cs

```diff
@@ -160,7 +165,71 @@ private static void ReadMetadataValue(ManifestMetadata manifestMetadata, XElemen
                 {
                     throw new InvalidDataException(string.Format(NuGetResources.Manifest_PropertyValueReadFailure, value, element.Name.LocalName), ex);
                 }
-                
+            }
+        }
+
+        private static LicenseMetadata ReadLicenseMetadata(XElement licenseNode)
```

Comment:
why have a separate implementation here? https://github.com/NuGet/NuGet.Client/pull/2450/files#diff-1b9e79be49dee677340ad8cdbc5399e0R416 

why can't the logic be refactored to read metadata from an XElement and be moved to a common place?

Reply:
The manifest reader is stricter since it's used on pack. 

Since this is arguably not a package breaker, the NuspecReader should be more permissive. 
This implementation is also driven by the consumption.

Reply:
The LicenseUrl/ProjectUrl failures in parsing were ignored before. We keep the same behavior. Will consult with Rob and the PMs.

Reply:
`Since this is arguably not a package breaker, the NuspecReader should be more permissive.
This implementation is also driven by the consumption.`

Have we capture these differences between pack vs UI consumption? Couldn't find relevant details in spec Also, I'm not sure I fully understand this discrimination... If I'm not allowing to pack something, then how does it matter that I'm more lenient at consumption?

Reply:
Explanation in the other comment about why we are more lenient on consumption. 
Also we don't have a monopoly over pack and people are free to edit their package manually as well. 

The NuspecReader is only used during consumption. 
The manifest reader is used during pack. This is more about how pack is written than anything spec related.